hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/contracts/operations/HederaExtCodeHashOperation.java

```diff
@@ -59,17 +60,25 @@
 public class HederaExtCodeHashOperation extends ExtCodeHashOperation {
 
     private final BiPredicate<Address, MessageFrame> addressValidator;
+    private final Predicate<Address> systemAccountDetector;
 
     public HederaExtCodeHashOperation(
-            GasCalculator gasCalculator, BiPredicate<Address, MessageFrame> addressValidator) {
+            GasCalculator gasCalculator,
+            BiPredicate<Address, MessageFrame> addressValidator,
+            Predicate<Address> systemAccountDetector) {
         super(gasCalculator);
         this.addressValidator = addressValidator;
+        this.systemAccountDetector = systemAccountDetector;
     }
 
     @Override
     public OperationResult execute(MessageFrame frame, EVM evm) {
         try {
             final Address address = Words.toAddress(frame.popStackItem());
+            if (systemAccountDetector.test(address)) {
```

Comment:
This is so much easier to reason about then whatever `HederaEvmOperationsUtil.addressCheckExecution` does.